### PR TITLE
Better support SARIF ruleId attribute

### PR DESCRIPTION
**Pull request by damiencarol**
According to specification, attribute `ruleId` could be missing from reports. (cf https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317643 )

> Not all existing analysis tools emit the equivalent of a ruleId in their output. 

This PR allow to import reports for tools that don't emit `ruleId` properly (like Gitleaks `7.5.0` which we use for a new unit tests) 

### Work done
- [x] better support attribute `ruleId` when tools doesn't set it cleanly (missing, reference but no data, etc...)
- [x] implement `mitigation` attribute

### CODE REVIEW

File: dojo/tools/sarif/parser.py [hunk 1]
```diff
@@ -185,18 +196,19 @@ def get_item(result, rules, artifacts, run_date):
     # we add a special 'None' case if there is no CWE
     cwes = [0]
     if rule is not None:
```
Comment (StefanFl) [inline dojo/tools/sarif/parser.py:198]: You have 2 times `if rule is not None:`, here and in line 179. Can you put the code under one `if`?

File: dojo/tools/sarif/parser.py [hunk 2]
```diff
@@ -185,18 +196,19 @@ def get_item(result, rules, artifacts, run_date):
...
-        cwes_extracted = get_rule_cwes(result)
+        cwes_extracted = get_rule_cwes(rule)
+        cwes_properties_extracted = get_result_cwes_properties(result)
```
Comment (StefanFl) [inline dojo/tools/sarif/parser.py:200]: `get_result_cwes_properties` is now only called when there are rules. Wouldn't it make sense to get CWEs from the results if there are no rules as well?

**Review by StefanFl — COMMENTED** [review]
See my 2 questions.

File: dojo/tools/sarif/parser.py (hunk 1 above)
Comment (damiencarol) [reply dojo/tools/sarif/parser.py]: modified.

File: dojo/tools/sarif/parser.py (hunk 2 above)
Comment (damiencarol) [reply dojo/tools/sarif/parser.py]: Modified following your comment

**Comment by damiencarol** [comment]
@StefanFl addressed your comments and also implemented `mitigation` attribute.

**Review by StefanFl — APPROVED** [review]

**Review by valentijnscholten — APPROVED** [review]

**Pull request merged**